### PR TITLE
Add Stamp Duty

### PR DESCRIPTION
**Pull request by nikhilwoodruff**

### CODE REVIEW

File: openfisca_uk/parameters/hmrc/stamp_duty/non_residential/purchase.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      2003-07-10: 250_000
+  - rate:
+      2003-07-10: 0.04
+      2016-09-15: null
+    threshold:
+      2003-07-10: 500_000
+      2016-09-15: null
+metadata:
+  threshold_unit: currency-GBP
+  rate_unit: currency-GBP
```
Comment (MaxGhenis) [inline openfisca_uk/parameters/hmrc/stamp_duty/non_residential/purchase.yaml:25]:
```suggestion
  rate_unit: /1
```

File: openfisca_uk/parameters/hmrc/stamp_duty/non_residential/rent.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      2003-07-10: 0.01
+    threshold:
+      2003-07-10: 150_000
+  - rate:
+      2016-09-15: 0.02
+    threshold:
+      2016-09-15: 5_000_000
+metadata:
+  threshold_unit: currency-GBP
+  rate_unit: currency-GBP
```
Comment (MaxGhenis) [inline openfisca_uk/parameters/hmrc/stamp_duty/non_residential/rent.yaml:17]:
```suggestion
  rate_unit: /1
```

File: openfisca_uk/variables/hmrc/stamp_duty.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        )
+
+
+class stamp_duty_liable(Variable):
+    label = "Liable for Stamp Duty"
+    documentation = "Whether the household is liable for Stamp Duty Land Tax"
+    entity = Household
+    definition_period = YEAR
+    value_type = bool
+    unit = "currency-GBP"
```
Comment (MaxGhenis) [inline openfisca_uk/variables/hmrc/stamp_duty.py:86]: Skip?

File: openfisca_uk/variables/hmrc/stamp_duty.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+    def formula(household, period):
+        country = household("country", period)
+        countries = country.possible_values
+        return (country == countries.ENGLAND) | (
```
Comment (MaxGhenis) [inline openfisca_uk/variables/hmrc/stamp_duty.py:91]: Can this use `isin`?

File: openfisca_uk/variables/hmrc/stamp_duty.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+            0,
+            second_home_price,
+        )
+        additional_residential_purchase_tax = (
+            stamp_duty.residential.purchase.additional.rate.calc(price)
+        )
+        # Tax on residential rents
+        cumulative_rent = household("cumulative_residential_rent", period)
+        rent = household("rent", period)
+        residential_rent_tax = stamp_duty.residential.rent.calc(
```
Comment (MaxGhenis) [inline openfisca_uk/variables/hmrc/stamp_duty.py:42]: I haven't seen this notation before - parameters can call `calc`?

File: openfisca_uk/variables/hmrc/stamp_duty.py [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    label = "Stamp Duty Land Tax"
+    documentation = "Total tax liability for Stamp Duty Land Tax"
+    entity = Household
+    definition_period = YEAR
+    value_type = float
+    unit = "currency-GBP"
+
+    def formula(household, period):
+
+        total_stamp_duty = household(
```
Comment (MaxGhenis) [inline openfisca_uk/variables/hmrc/stamp_duty.py:106]: skip

**Review by MaxGhenis — COMMENTED** [review]
I'll review the logic in more detail tomorrow

File: openfisca_uk/variables/hmrc/stamp_duty.py (hunk 5 above)
Comment (nikhilwoodruff) [reply openfisca_uk/variables/hmrc/stamp_duty.py:42]: It's the ParameterScale interface - used in rates, etc. in tax calculations too

**Pull request merged**